packages/extension/src/ui/features/multisig/TransactionConfirmationsScreen.tsx

```diff
@@ -0,0 +1,144 @@
+import {
+  BarBackButton,
+  H4,
+  H6,
+  NavigationContainer,
+  P3,
+  icons,
+} from "@argent/ui"
+import { Box, Divider, Flex } from "@chakra-ui/react"
+import { useMemo } from "react"
+
+import { useRouteRequestId } from "../../routes"
+import { formatTruncatedSignerKey } from "../../services/addresses"
+import { transformTransaction } from "../accountActivity/transform"
+import { getTransactionFromPendingMultisigTransaction } from "../accountActivity/transform/transaction/transformers/pendingMultisigTransactionAdapter"
+import { Account } from "../accounts/Account"
+import { useEncodedPublicKeys, useSignerKey } from "../accounts/usePublicKey"
+import { useRouteAccount } from "../shield/useRouteAccount"
+import { useMultisigInfo } from "./hooks/useMultisigInfo"
+import { useMultisigPendingTransaction } from "./hooks/useMultisigPendingTransaction"
+import { useMultisigRequest } from "./hooks/useMultisigRequest"
+
+const { TickIcon } = icons
+export const TransactionConfirmationsScreen = () => {
+  const selectedAccount = useRouteAccount()
+  const requestId = useRouteRequestId()
+  const { data: pendingTransaction } = useMultisigPendingTransaction(requestId)
+  const { data } = useMultisigRequest({
+    account: selectedAccount,
+    requestId,
+  })
+  const signerKey = useSignerKey()
+
+  const transactionTransformed = useMemo(() => {
+    if (data && pendingTransaction && selectedAccount) {
+      return transformTransaction({
+        transaction: getTransactionFromPendingMultisigTransaction(
+          {
+            ...pendingTransaction,
+            data,
+          },
+          selectedAccount,
+        ),
+        accountAddress: selectedAccount.address,
+      })
+    }
+  }, [data, pendingTransaction, selectedAccount])
+  const approvedSignersPublicKey = useEncodedPublicKeys(
+    data?.content.approvedSigners ?? [],
+  )
+
+  const nonApprovedSignersPublicKey = useEncodedPublicKeys(
+    data?.content.nonApprovedSigners ?? [],
+  )
+  return (
+    <NavigationContainer
+      leftButton={<BarBackButton />}
+      title={transactionTransformed?.displayName}
+    >
+      <Box p={4}>
+        <H4>Waiting for confirmations...</H4>
+        {selectedAccount && data && (
+          <TransactionConfirmationsScreenSubtitle
+            account={selectedAccount}
+            approvedSignersLength={data?.content.approvedSigners.length}
+          />
+        )}
+        <Divider my={4} color="neutrals.800" />
+
+        <P3 color="neutrals.300" mb={3}>
+          Me
+        </P3>
+        <Box borderRadius="lg" backgroundColor="neutrals.800" p={4} mb={3}>
+          {signerKey && (
+            <Flex alignItems="center" justifyContent="space-between">
+              <H6 color="white">{formatTruncatedSignerKey(signerKey)} </H6>
+              {approvedSignersPublicKey.find((key) => key === signerKey) && (
```

Comment:
```suggestion
              {approvedSignersPublicKey.some((key) => key === signerKey) && (
```
I would use `some` here but `find` works too